app/reports/signatures_report.rb

```diff
@@ -24,16 +24,32 @@ def csv_headers
   # @private
   #
   # @return [ActiveRecord::Relation]
-  def pull_join
-    start_time = begin_date.in_time_zone.beginning_of_day
-    end_time   = end_date.in_time_zone.end_of_day
+  def events_in_range(start_time, end_time)
+    ShiftEvent
+      .includes(shift: [:work_site, :volunteer])
+      .where(occurred_at: start_time..end_time,
+             work_sites:  { active: true })
+      .order(:occurred_at)
+  end
 
+  def events_in_range_at_site(start_time, end_time, site_id)
     ShiftEvent
+      .joins(shift: [:work_site, :volunteer])
+      .where(occurred_at: start_time..end_time,
+             shifts:      { work_site_id:     site_id })
       .includes(shift: [:work_site, :volunteer])
-      .where(occurred_at: start_time..end_time)
       .order(:occurred_at)
   end
 
+  def pull_join(site_id = 0)
```

Comment:
Our database is zero-indexed, so `site_id = 0` might be a valid site. I'd set the fallback to `-1` or something similar so that all valid sites are reachable.